core/src/main/com/ruslan/entity/BaseEntity.java

```diff
@@ -0,0 +1,30 @@
+package com.ruslan.entity;
+
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.MappedSuperclass;
+
+import java.io.Serializable;
+
+@MappedSuperclass
+public abstract class BaseEntity<T extends Serializable> {
```

Comment:
you can use Lombok annotations here as well

Reply:
added